### ISSUE TITLE
Enable key repeat for non text keys

### ISSUE THREAD

**Issue by mkrueger** [emilk/egui#2387]
I'm currently developing a terminal application using egui. I use a fragment shader to display the content and use it inside a ViewPort scroll area:

```
           let output = ScrollArea::vertical()
           .auto_shrink([false; 2])
           .stick_to_bottom(true)
           .show_viewport(ui, |ui, viewport| { …

            let events = ui.input().events.clone();
            for e in &events { …
            }
```

I need to handle the key events myself and it works quite well - however key repeats for the command keys Arrow/Backspace are not working - only the Text event does. I can't figure out how to detect key repeats the `egui::Event::Key { key, pressed: true, modifiers }`  event is only once in the list.

Another issue is - can I disable the scroll drag in the ScrollArea somehow? It's interfering with my mouse selection in the control.

**Comment by emilk** [comment]
Starting in https://github.com/emilk/egui/pull/2334 (recently), key repeates are ignored. Perhaps this was a bad idea.

> Another issue is - can I disable the scroll drag in the ScrollArea somehow? It's interfering with my mouse selection in the control.

See https://github.com/emilk/egui/pull/2163

**Comment by mkrueger** [comment]
It's not a bad idea per se having events for key press and release without repeat does make sense. 
But I would vote for maybe a KeyReceived event for the key repeats.

The agree that the frame based event system is a bit tricky. But I really, really need key repeats here :/ (however now I know where to patch that - thanks.)

**Comment by mkrueger** [comment]
@emilk: Something like that would work https://github.com/emilk/egui/pull/2389

However I can't test that locally - my application crashes 
```

thread 'main' panicked at 'failed to create glutin window surface: Error { raw_code: Some(12297), raw_os_message: None, kind: BadMatch }', /home/mkrueger/work/egui/crates/eframe/src/native/run.rs:408:18
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:142:14
   2: core::result::unwrap_failed
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/result.rs:1785:5
   3: core::result::Result<T,E>::expect
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/result.rs:1064:23
   4: eframe::native::run::glow_integration::GlutinWindowContext::new
             at /home/mkrueger/work/egui/crates/eframe/src/native/run.rs:406:30
   5: eframe::native::run::glow_integration::GlowWinitApp::create_glutin_windowed_context
             at /home/mkrueger/work/egui/crates/eframe/src/native/run.rs:504:26
   6: eframe::native::run::glow_integration::GlowWinitApp::init_run_state
             at /home/mkrueger/work/egui/crates/eframe/src/native/run.rs:520:35
   7: <eframe::native::run::glow_integration::GlowWinitApp as eframe::native::run::WinitApp>::on_event
             at /home/mkrueger/work/egui/crates/eframe/src/native/run.rs:732:25
   8: eframe::native::run::run_and_return::{{closure}}
             at /home/mkrueger/work/egui/crates/eframe/src/native/run.rs:140:22
   9: winit::platform_impl::platform::sticky_exit_callback
             at /home/mkrueger/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.27.5/src/platform_impl/linux/mod.rs:849:9
  10: winit::platform_impl::platform::x11::EventLoop<T>::run_return::single_iteration
             at /home/mkrueger/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.27.5/src/platform_impl/linux/x11/mod.rs:339:17
  11: winit::platform_impl::platform::x11::EventLoop<T>::run_return
             at /home/mkrueger/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.27.5/src/platform_impl/linux/x11/mod.rs:448:31
  12: winit::platform_impl::platform::EventLoop<T>::run_return
             at /home/mkrueger/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.27.5/src/platform_impl/linux/mod.rs:748:56
  13: <winit::event_loop::EventLoop<T> as winit::platform::run_return::EventLoopExtRunReturn>::run_return
             at /home/mkrueger/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.27.5/src/platform/run_return.rs:62:9
  14: eframe::native::run::run_and_return
             at /home/mkrueger/work/egui/crates/eframe/src/native/run.rs:107:5
  15: eframe::native::run::glow_integration::run_glow::{{closure}}
             at /home/mkrueger/work/egui/crates/eframe/src/native/run.rs:851:17
  16: eframe::native::run::with_event_loop::{{closure}}
             at /home/mkrueger/work/egui/crates/eframe/src/native/run.rs:96:9
  17: std::thread::local::LocalKey<T>::try_with
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/thread/local.rs:445:16
  18: std::thread::local::LocalKey<T>::with
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/thread/local.rs:421:9
  19: eframe::native::run::with_event_loop
             at /home/mkrueger/work/egui/crates/eframe/src/native/run.rs:89:5
  20: eframe::native::run::glow_integration::run_glow
             at /home/mkrueger/work/egui/crates/eframe/src/native/run.rs:844:13
  21: eframe::run_native
             at /home/mkrueger/work/egui/crates/eframe/src/lib.rs:181:13
  22: icy_term::main::{{closure}}
             at ./src/main.rs:33:5
  23: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/future/mod.rs:91:19
  24: tokio::park::thread::CachedParkThread::block_on::{{closure}}
             at /home/mkrueger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/park/thread.rs:267:54
  25: tokio::coop::with_budget::{{closure}}
             at /home/mkrueger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/coop.rs:102:9
  26: std::thread::local::LocalKey<T>::try_with
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/thread/local.rs:445:16
  27: std::thread::local::LocalKey<T>::with
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/thread/local.rs:421:9
  28: tokio::coop::with_budget
             at /home/mkrueger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/coop.rs:95:5
  29: tokio::coop::budget
             at /home/mkrueger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/coop.rs:72:5
  30: tokio::park::thread::CachedParkThread::block_on
             at /home/mkrueger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/park/thread.rs:267:31
  31: tokio::runtime::enter::Enter::block_on
             at /home/mkrueger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/runtime/enter.rs:152:13
  32: tokio::runtime::scheduler::multi_thread::MultiThread::block_on
             at /home/mkrueger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/runtime/scheduler/multi_thread/mod.rs:79:9
  33: tokio::runtime::Runtime::block_on
             at /home/mkrueger/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/runtime/mod.rs:492:44
  34: icy_term::main
             at ./src/main.rs:33:5
  35: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
```
